### PR TITLE
 roe-2573 update createNewSubmission

### PR DESCRIPTION
**Pull request by lduranteau**
### JIRA link
https://companieshouse.atlassian.net/browse/ROE-2573

### Change description
Move /transactions/{TRANSACTION_ID_KEY}/overseas-entity/start logic and remove the endpoint
 - move logic from createNewSubmissionForSaveAndResume to createNewSubmission
 - remove createNewSubmissionForSaveAndResume

### Work checklist

- [x] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Review by markpit — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesService.java [hunk 1]
```diff
@@ -87,30 +87,11 @@ SubmissionType getSubmissionType(String requestId, String overseasEntityId) thro
...
-    }
-
-    private ResponseEntity<Object> createOverseasEntity(Transaction transaction,
                                                         OverseasEntitySubmissionDto overseasEntitySubmissionDto,
                                                         String requestId,
-                                                        String userId,
-                                                        boolean addResumeLinkToTransaction) throws ServiceException {
-        ApiLogger.debugContext(requestId, "Called createOverseasEntity(...), with 'add resume link' = " + addResumeLinkToTransaction);
+                                                        String userId) throws ServiceException {
+        ApiLogger.debugContext(requestId, "Called createOverseasEntity(...), with 'add resume link'");
```
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesService.java:94]: I think the _, with 'add resume link'_ part can be removed from the message.

Comment (lduranteau) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesService.java]: Removed

**Review by mwestacott — COMMENTED** [review]
LGTM

**Review by dlloyd1-ch — APPROVED** [review]
LGTM but can't be merged until all the sub tasks in this story are complete.

**Review by mwestacott — APPROVED** [review]
Can't merge yet though.

**Pull request merged**